ocaml/src/nsm_host_plugin.ml

```diff
@@ -375,6 +378,24 @@ let handle_query : type i o. read_user_db -> (i, o) Nsm_host_protocol.Protocol.q
         try Result.Ok (nsm_query tag namespace_id req)
         with Nsm_model.Err.Nsm_exn (err, _) -> Result.Error err)
        req
+  | UpdateSession ->
+     begin
+       List.iter
+         (function
+          | "manifest_ser", vo ->
+             begin
+               match vo with
+               | None -> ()
+               | Some v ->
+                  let new_version = deserialize Llio.int8_from v in
+                  Nsm_protocol.Session.set_manifest_ser session new_version
```

Comment:
What are the exact semantics of the manifest version that is requested?
Is it that the server _should_ give back a manifest serialized with this version, or that it _can_ give back a manifest up to the specified version?
In case of the former an error should be returned when requesting a newer version than known by the server, in case of the latter the code here should read `(min 2 new_version)`.
As it is right now a newer (as of yet unexisting client) will have problems when explaining to an old server that it wants manifests in version format 3.

(My preference goes towards the "it can give back a manifest up to the specified version" semantics.)